docs/guides/metrics.md

```diff
@@ -0,0 +1,156 @@
+
+**Component Metrics**: The core Dynamo backend system automatically exposes metrics with the `dynamo_component_*` prefix. The followings are examples that currently exist for all components that use the `DistributedRuntime` framework. More metrics are being added and will appear in future releases:
+
+- `dynamo_component_concurrent_requests`: Requests currently being processed (gauge)
+- `dynamo_component_request_bytes_total`: Total bytes received in requests (counter)
+- `dynamo_component_request_duration_seconds`: Request processing time (histogram)
+- `dynamo_component_requests_total`: Total requests processed (counter)
+- `dynamo_component_response_bytes_total`: Total bytes sent in responses (counter)
+- `dynamo_component_system_uptime_seconds`: DistributedRuntime uptime (gauge)
+
+**Specialized Component Metrics**: Some components expose additional metrics specific to their functionality. For example, the Preprocessor will have labels in the following convention:
+
+- `dynamo_preprocessor_*`: Metrics specific to preprocessor components
+
+**Frontend Metrics**: When using Dynamo HTTP Frontend (`--framework VLLM` or `--framework TENSORRTLLM`), these metrics are automatically exposed with the `dynamo_frontend_*` prefix. These metrics include `model` labels containing the model name:
+
+- `dynamo_frontend_inflight_requests`: Inflight requests (gauge)
+- `dynamo_frontend_input_sequence_tokens`: Input sequence length (histogram)
+- `dynamo_frontend_inter_token_latency_seconds`: Inter-token latency (histogram)
+- `dynamo_frontend_output_sequence_tokens`: Output sequence length (histogram)
+- `dynamo_frontend_request_duration_seconds`: LLM request duration (histogram)
+- `dynamo_frontend_requests_total`: Total LLM requests (counter)
+- `dynamo_frontend_time_to_first_token_seconds`: Time to first token (histogram)
+
+## Metrics Hierarchy
+
+The `MetricsRegistry` trait is implemented by `DistributedRuntime`, `Namespace`, `Component`, and `Endpoint`, providing a hierarchical approach to metric collection that matches Dynamo's distributed architecture:
+
+- `DistributedRuntime`: Global metrics across the entire runtime
+- `Namespace`: Metrics scoped to a specific dynamo_namespace
+- `Component`: Metrics for a specific dynamo_component within a namespace
+- `Endpoint`: Metrics for individual dynamo_endpoint within a component
+
+This hierarchical structure allows you to create metrics at the appropriate level of granularity for your monitoring needs.
+
+## Environment Configuration
+
+Enable metrics with environment variables:
+
+```bash
+export DYN_SYSTEM_ENABLED=true
+export DYN_SYSTEM_PORT=8081  # Use 0 for random port
+```
+
+With `DYN_SYSTEM_ENABLED=true`, Dynamo automatically adds:
+- `dynamo_system_uptime_seconds`: System uptime counter
+- HTTP server metrics for the metrics endpoint
```

Comment:
Is DYN_SYSTEM_ENABLED also correlated with the "automatic" `dynamo_component_*` metrics? or unrelated?